src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -0,0 +1,157 @@
+
+        Important: This needs to be a unique ID and return the same value for
+        all steps of a pipeline run.
+
+        Returns:
+            The orchestrator run id.
+        """
+        print(os.environ)
+        try:
+            return os.environ[ENV_ZENML_SAGEMAKER_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_SAGEMAKER_RUN_ID}."
+            )
+
+    def prepare_pipeline_deployment(
+        self, deployment: "PipelineDeployment", stack: "Stack"
+    ) -> None:
+        """Build a Docker image and push it to the container registry.
+
+        Args:
+            deployment: The pipeline deployment configuration.
+            stack: The stack on which the pipeline will be deployed.
+        """
+        docker_image_builder = PipelineDockerImageBuilder()
+        repo_digest = docker_image_builder.build_and_push_docker_image(
+            deployment=deployment, stack=stack
+        )
+        # repo_digest = "715803424590.dkr.ecr.eu-north-1.amazonaws.com/zenml:not_so_basic_pipeline"
+        deployment.add_extra(ORCHESTRATOR_DOCKER_IMAGE_KEY, repo_digest)
+
+    def _get_timestamp(self) -> str:
+        """Returns the current timestamp."""
+        return datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+
+    def _get_unique_run_id(self, deployment: "PipelineDeployment") -> str:
+        """Returns a unique run id for the pipeline run.
+
+        Args:
+            deployment: The pipeline deployment configuration.
+
+        Returns:
+            A unique run id.
+        """
+        return str(deployment.pipeline_id) + self._get_timestamp()
+
+    def prepare_or_run_pipeline(
+        self, deployment: "PipelineDeployment", stack: "Stack"
+    ) -> Any:
+        """Prepares or runs a pipeline on Sagemaker.
+
+        Args:
+            deployment: The deployment to prepare or run.
+            stack: The stack to run on.
+
+        Returns:
+            The result of the pipeline run.
+        """
+        session = sagemaker.Session(default_bucket=self.config.bucket)
+        image_name = deployment.pipeline.extra[ORCHESTRATOR_DOCKER_IMAGE_KEY]
+        sagemaker_steps = []
+        for step_name, step in deployment.steps.items():
+            command = StepEntrypointConfiguration.get_entrypoint_command()
+            arguments = StepEntrypointConfiguration.get_entrypoint_arguments(
+                step_name=step_name
+            )
+            entrypoint = command + arguments
+
+            processor = sagemaker.processing.Processor(
+                role=sagemaker.get_execution_role(),
+                image_uri=image_name,
+                instance_count=1,
+                instance_type=self.config.instance_type,
+                entrypoint=entrypoint,
+                sagemaker_session=session,
+                base_job_name=deployment.run_name,
+                env={
```

Comment:
We can give the ability to pass in more things in the env here via settings and config